kahuna/public/js/leases/leases.js

```diff
@@ -87,31 +94,56 @@ leases.controller('LeasesCtrl', [
                     : "This image already has a syndication lease. It will be overwritten. " +
                       "Do you wish to proceed?";
                   const shouldApplyLeases = $window.confirm(confirmText);
-                  if (!shouldApplyLeases) {
+                    if (!shouldApplyLeases) {
+                      //We're not saving it if the user cancels, so clear up
+                    //   return ctrl.updateLeases();
+                    ctrl.adding = false;
                     return;
                   }
                 }
 
+
+
               leaseService.batchAdd(ctrl.newLease, ctrl.images)
                 .catch((e) => {
                   console.error(e);
                   alertFailed('Something went wrong when saving, please try again.');
                 })
-                .finally(() => {
+                  .finally(() => {
+                      console.log("HELO");
                     ctrl.resetLeaseForm();
                 });
             }
         };
 
         // These events allow this control to work as a hybrid on the upload page
+        // Update from future: not very well.
         const batchAddLeasesEvent = 'events:batch-apply:add-leases';
         const batchRemoveLeasesEvent = 'events:batch-apply:remove-leases';
 
         if (Boolean(ctrl.withBatch)) {
-            $scope.$on(batchAddLeasesEvent,
-                    (e, leases) => leaseService.replace(ctrl.images[0], leases));
+          $scope.$on(batchAddLeasesEvent,
+              (e, leases) => {
+                  ctrl.receivingBatch = true;
+                  leaseService.replace(ctrl.images[0], leases).then((a) => {
+                //These get run before the lease is shown in the UI. Fix.
```

Comment:
Is this fix due in this PR?

Reply:
Ooops, that comment doesn't belong any more. 

When `replace` is run, it triggers the `leasesUpdated` event which refreshes all the leases on the page, which means we can just wait for that. (Will update comment)

Reply:
Deleted